metrics_utility/test/test_extra_params_validation.py

```diff
@@ -0,0 +1,123 @@
+import os
+
+import pytest
+
+from openpyxl import load_workbook
+from pandas import read_excel
+
+from metrics_utility.test.util import run_build_int
+
+
+env_vars = {
+    'METRICS_UTILITY_SHIP_PATH': './metrics_utility/test/test_data',
+    'METRICS_UTILITY_SHIP_TARGET': 'directory',
+    'METRICS_UTILITY_REPORT_TYPE': 'CCSPv2',
+    'METRICS_UTILITY_OPTIONAL_CCSP_REPORT_SHEETS': 'ccsp_summary,managed_nodes,indirectly_managed_nodes,usage_by_organizations',
+}
+file_path = './metrics_utility/test/test_data/reports/2025/02/CCSPv2-2025-02-25--2025-02-26.xlsx'
+
+
+def test_uses_since_and_until_values():
+    env_vars['METRICS_UTILITY_REPORT_TYPE'] = 'CCSPv2'
+    args = {'since': '2025-02-25', 'until': '2025-02-26', 'force': True}
+    run_build_int(env_vars, args)
+    try:
+        workbook = load_workbook(filename=file_path)
+        worksheet = read_excel(file_path, sheet_name='Usage Reporting')
+        cell_value = worksheet.iat[3, 1]
+
+        assert cell_value == '2025-02-25, 2025-02-26', f"Expected '2025-02-25, 2025-02-26' but got '{cell_value}'"
+    finally:
+        workbook.close()
+        os.remove(file_path)
+
+
+month_file_path = './metrics_utility/test/test_data/reports/2025/04/CCSPv2-2025-04.xlsx'
+
+
+def test_accepts_month_and_ignores_since_until():
+    env_vars['METRICS_UTILITY_REPORT_TYPE'] = 'CCSPv2'
+    args = {'since': '2024-01-01', 'until': '2024-02-01', 'month': '2025-04', 'force': True}
+    # Build report, then assert that the report contains only dates of the month provided
+    run_build_int(env_vars, args)
+    try:
+        workbook = load_workbook(filename=month_file_path)
+        worksheet = read_excel(month_file_path, sheet_name='Usage Reporting')
+        cell_value = worksheet.iat[3, 1]
+        assert cell_value == '2025-04', f"Expected '2025-04' but got '{cell_value}'"
+    finally:
+        workbook.close()
+        os.remove(month_file_path)
+
+
+def test_renewal_guidancev2_fails_with_until_params(caplog):
+    command_args = [
+        {'args': {'until': '2025-01-01'}, 'report_type': 'RENEWAL_GUIDANCE'},
+        {'args': {'until': '2025-01-01'}, 'report_type': 'RENEWAL_GUIDANCEv2'},
+        {'args': {'until': '2025-01-01', 'since': '2024-02-01'}, 'report_type': 'RENEWAL_GUIDANCE'},
+    ]
+    for arg in command_args:
+        handle_exception(
+            caplog,
+            {**env_vars, 'METRICS_UTILITY_REPORT_TYPE': arg['report_type']},
+            arg['args'],
+            'The --until parameter is not allowed when environment variable METRICS_UTILITY_REPORT_TYPE is RENEWAL_GUIDANCE or RENEWAL_GUIDANCEv2',
+        )
+
+
+def test_invalid_month_format(caplog):
+    bad_inputs = ['abc', '12', 'mo3', '3mon', '3months', '3m', '3mo']
+
+    for bad_input in bad_inputs:
+        handle_exception(caplog, env_vars, {'month': bad_input}, 'Invalid --month format. Supported date format: YYYY-MM')
+
+
+def handle_exception(caplog, env_vars, params, error_message):
```

Comment:
this will not be needed if the test just calls _handle, and tests the right exception was returned